Tests/Functional/Token/EmailWithCustomObjectDynamicContentFunctionalTest.php

```diff
@@ -0,0 +1,189 @@
+
+        $this->customItemModel       = self::$container->get('mautic.custom.model.item');
+        $this->customFieldValueModel = self::$container->get('mautic.custom.model.field.value');
+        $this->emailModel            = self::$container->get('mautic.email.model.email');
+        $this->emailStatRepository   = $this->em->getRepository(Stat::class);
+    }
+
+    public function testEmailWithCustomObjectDynamicContent(): void
+    {
+        $customObject = $this->createCustomObjectWithAllFields(self::$container, 'Car');
+        $customItem   = new CustomItem($customObject);
+
+        $customItem->setName('Nexon');
+        $this->customFieldValueModel->createValuesForItem($customItem);
+
+        $textValue       = $customItem->findCustomFieldValueForFieldAlias('text-test-field');
+        $textValue->setValue('Tata');
+
+        $customItem = $this->customItemModel->save($customItem);
+
+        $lead1  = $this->createLead('nexon@tata.com');
+        $lead2  = $this->createLead('noitem@tata.com');
+        $email  = $this->createEmail();
+        $email->setDynamicContent(
+            [
+                [
+                    'tokenName' => 'Dynamic Content 1',
+                    'content'   => 'Default Dynamic Content',
+                    'filters'   => [
+                        [
+                            'content' => null,
+                            'filters' => [
+                            ],
+                        ],
+                    ],
+                ],
+                [
+                    'tokenName' => 'Dynamic Content 2',
+                    'content'   => 'Default Dynamic Content',
+                    'filters'   => [
+                        [
+                            'content' => 'Nexon Dynamic Content',
+                            'filters' => [
+                                [
+                                    'glue'     => 'and',
+                                    'field'    => 'cmf_'.$textValue->getCustomField()->getId(),
+                                    'object'   => 'custom_object',
+                                    'type'     => 'text',
+                                    'filter'   => 'Tata',
+                                    'display'  => $customObject->getAlias().': text-test-field',
+                                    'operator' => '=',
+                                ],
+                                [
+                                    'glue'     => 'and',
+                                    'field'    => 'cmo_'.$customObject->getId(),
+                                    'object'   => 'custom_object',
+                                    'type'     => 'text',
+                                    'filter'   => 'Nexon',
+                                    'display'  => $customObject->getAlias(),
+                                    'operator' => '=',
+                                ],
+                            ],
+                        ],
```

Comment:
Please use data provider to cover multiple combinations of the filters and operators.